server/storage/database_test.go

```diff
@@ -2,290 +2,326 @@ package storage
 
 import (
 	"database/sql"
+	"io/ioutil"
+	"os"
 	"testing"
 
-	"github.com/DATA-DOG/go-sqlmock"
-	"github.com/go-sql-driver/mysql"
+	"github.com/endophage/gotuf/data"
+	"github.com/jinzhu/gorm"
+	_ "github.com/mattn/go-sqlite3"
 	"github.com/stretchr/testify/assert"
 )
 
-func TestMySQLUpdateCurrent(t *testing.T) {
-	db, err := sqlmock.New()
-	assert.Nil(t, err, "Could not initialize mock DB")
-	s := NewMySQLStorage(db)
-	update := MetaUpdate{
-		Role:    "root",
-		Version: 0,
-		Data:    []byte("1"),
-	}
-	sqlmock.ExpectQuery("SELECT count\\(\\*\\) FROM `tuf_files` WHERE `gun`=\\? AND `role`=\\? AND `version`>=\\?\\;").WithArgs(
-		"testGUN",
-		update.Role,
-		update.Version,
-	).WillReturnRows(sqlmock.RowsFromCSVString([]string{"count(*)"}, "0"))
-	sqlmock.ExpectExec("INSERT INTO `tuf_files` \\(`gun`, `role`, `version`, `data`\\) VALUES \\(\\?,\\?,\\?,\\?\\);").WithArgs(
-		"testGUN",
-		update.Role,
-		update.Version,
-		update.Data,
-	).WillReturnResult(sqlmock.NewResult(0, 1))
-
-	err = s.UpdateCurrent(
-		"testGUN",
-		update,
-	)
-	assert.Nil(t, err, "UpdateCurrent errored unexpectedly: %v", err)
-
-	// There's a bug in the mock lib
-	//err = db.Close()
-	//assert.Nil(t, err, "Expectation not met: %v", err)
+// GormTUFFile represents a TUF file in the database
+type GormTUFFile struct {
```

Comment:
Given we'll want to use the models when we do the full GORM integration, it would be good to put them in their own non-test file within the storage package. If it doesn't conflict with any else in the storage package, it would also be good to drop the `Gorm` prefix.

Rather than putting an `ID` field in, let's use the [`gorm.Model`](https://github.com/jinzhu/gorm#conventions) embed and fix the databases in stage/prod to have the `created_at`, `updated_at`, and `deleted_at` columns.